UnitTests/CommonTestUtils/AsyncTestHelper.cs

```diff
@@ -29,7 +29,7 @@ public static void WaitForPendingOperations()
         {
             try
             {
-                ThreadHelper.JoinableTaskContext?.Factory.Run(() => ThreadHelper.JoinPendingOperationsAsync());
+                ThreadHelper.JoinableTaskContext.Factory.Run(() => ThreadHelper.JoinPendingOperationsAsync());
```

Comment:
❗️ Do not change this. If an exception is thrown in `BeforeTest`, we will not be able to diagnose it because a different exception will be thrown in `AfterTest` and hide the cause.

Reply:
Valid point (but wrong place: `AsyncTestHelper.WaitForPendingOperations` is not used neither by `BeforeTest` nor by `AfterTest`.)

If a test case throws before a call `AsyncTestHelper.WaitForPendingOperations`, this function will not be called anymore.
And if `WaitForPendingOperations` is called and `ThreadHelper.JoinableTaskContext == null` then the test shall fail _always_ not randomly.

Regarding exceptions in `AfterTest`.
Maybe yes. Which exception takes precedence? The one thrown in `try { }` or the one thrown in `finally { }`?
So let's be explicit and add:
```
catch (Exception) if (_threadException != null)
{
    // ignore the follow-up exception
}
```

Reply:
`AsyncTestHelper.WaitForPendingOperations` previously returned immediately if the test didn't have a JoinableTaskContext (i.e. all work was complete because the scheduler wasn't even in use). That behavior seemed fine, but if we do change it, the method should throw `InvalidOperationException` instead of `NullReferenceException`.

Reply:
throws `InvalidOperationException` now if null